python/pyspark/sql/readwriter.py

```diff
@@ -733,11 +774,19 @@ def _test():
     import os
     import tempfile
     import py4j
+    import shutil
     from pyspark.context import SparkContext
     from pyspark.sql import SparkSession, Row
     import pyspark.sql.readwriter
 
-    os.chdir(os.environ["SPARK_HOME"])
+    spark_home = os.path.realpath(os.environ["SPARK_HOME"])
+
+    test_dir = tempfile.mkdtemp()
+    os.chdir(test_dir)
+
+    path = lambda x, y, z: os.path.join(x, y)
+
+    shutil.copytree(path(spark_home, 'python', 'test_support'), path(test_dir, 'python', 'test_support'))
```

Comment:
pep8 is saying this line is too long (over 100 chars) so its failing the style tests. You can run the style tests locally with `./dev/lint-python` as well for a faster turn around than Jenkins :)